<?xml version="1.0"?>
<robot name = "moving_platform">

  <material name="green">
    <color rgba= "0 1 0 1"/>
  </material> 
  
  <!-- Create world link -->
  <link name="world_link"/>
  
  <!-- Create moving platform -->
  <link name="moving_platform/base_link">
    <origin xyz="0 0 0" rpy="0 0 0" />
     <visual>
       <geometry>
         <box size="1 1 0.29" />
       </geometry>
    </visual>
    <inertial>
         <mass value="10"/>
      <inertia ixx="0.9083" ixy="0" ixz="0" iyy="0.9083" iyz="0" izz="1.66666"/>
    </inertial>
    <collision>
       <origin xyz="0 0 0" rpy="0 0 0"/>
       <geometry>
         <box size="1 1 0.29" />
       </geometry>
    </collision>
  </link>
  
  <gazebo reference="moving_platform/base_link">
   <material>Gazebo/White</material>
  </gazebo>

  <joint name="joint_world_base_link" type="fixed">
    <parent link="world_link"/>
    <child link="moving_platform/base_link"/>
    <origin xyz="0 0 0.15" rpy="0 0 0" />
  </joint>

  <gazebo>
  <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
    <robotNamespace>/moving_platform</robotNamespace>
  </plugin>
</gazebo>


  <!-- Create bumper to detect touchdown -->
 <joint name="joint_base_link_bumper_link" type="fixed">
    <parent link="moving_platform/base_link"/>
    <child link="moving_platform/bumper_link"/>
    <origin xyz="0 0 0.15" rpy="0 0 0" />
 </joint>

  <link name="moving_platform/bumper_link">
    <origin xyz="0 0 0" rpy="0 0 0" />
     <visual>
       <geometry>
         <box size="1 1 0.01" />
       </geometry>
    </visual>
    <collision>
       <origin xyz="0 0 0" rpy="0 0 0"/>
       <geometry>
         <box size="1 1 0.01" />
       </geometry>
    </collision>
  </link>

<gazebo reference="moving_platform/bumper_link">
   <material>Gazebo/Green</material>
</gazebo>

 <gazebo reference="moving_platform/bumper_link">
  <sensor name="main_bumper" type="contact">
  <selfCollide>true</selfCollide>
  <alwaysOn>true</alwaysOn>
  <updateRate>100.0</updateRate>
  <material>Gazebo/Red</material>
  <topic>__default_topic__</topic>
  <contact>
      <collision>world_link_fixed_joint_lump__moving_platform/bumper_link_collision_1</collision>
  </contact>
  <plugin name="gazebo_ros_bumper_controller" filename="libgazebo_ros_bumper.so">   
      <bumperTopicName>moving_platform/contact</bumperTopicName>
      <frameName>moving_platform/bumper_link</frameName>
      <alwaysOn>true</alwaysOn>
      <updateRate>100.0</updateRate>
  </plugin>
  </sensor>
</gazebo>

</robot>
